### PR TITLE
Fix a stupid bug on Replicated database recovery

### PR DESCRIPTION
**Pull request by tavplubix**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 9240a82c6481e3d1caf7d5c28e1e549d143a55e3 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51252/9240a82c6481e3d1caf7d5c28e1e549d143a55e3/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Databases/DatabaseReplicated.cpp [hunk 1]
```diff
@@ -710,8 +710,9 @@ BlockIO DatabaseReplicated::tryEnqueueReplicatedDDL(const ASTPtr & query, Contex
 
 static UUID getTableUUIDIfReplicated(const String & metadata, ContextPtr context)
 {
-    bool looks_like_replicated = metadata.find("ReplicatedMergeTree") != std::string::npos;
-    if (!looks_like_replicated)
+    bool looks_like_replicated = metadata.find("Replicated") != std::string::npos;
```
Comment (alexey-milovidov) [inline src/Databases/DatabaseReplicated.cpp:713]: The substring search looks strange.

CREATE TABLE test (Replicated UInt8) ENGINE = Log

Comment (tavplubix) [reply src/Databases/DatabaseReplicated.cpp:713]: There's a correct check a few lines below

Comment (nikitamikhaylov) [reply src/Databases/DatabaseReplicated.cpp:713]: Maybe just delete these lines, so query will be parsed each time?

**Review by alesapin — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
@tavplubix is it possible to add a test?

**Comment by tavplubix** [comment]
@alexey-milovidov, a test was added in https://github.com/ClickHouse/ClickHouse/pull/51284

**Pull request merged**